server/src/main/java/io/crate/analyze/TableElementsAnalyzer.java

```diff
@@ -619,8 +625,26 @@ private void processConstraint(RefBuilder builder, ColumnConstraint<Expression>
                 }
             } else if (constraint instanceof NotNullColumnConstraint<Expression>) {
                 builder.nullable = false;
+                if (builder.explicitNullable) {
+                    throw new IllegalArgumentException(String.format(Locale.ENGLISH, 
+                        "Column \"%s\" is declared NULL, therefore, cannot be declared NOT NULL", columnName));
+                }
             } else if (constraint instanceof PrimaryKeyColumnConstraint<Expression>) {
                 markAsPrimaryKey(builder);
+                if (builder.explicitNullable) {
+                    throw new IllegalArgumentException(String.format(Locale.ENGLISH, 
+                        "Column \"%s\" is declared NULL, therefore, cannot be declared as a PRIMARY KEY", columnName));
+                }
```

Comment:
this is redundant since we already have such check in `markAsPrimaryKey` which is called for both column/table level PK definitions.

Reply:
good catch. taking it out